File: explorations/globalString.R
library(Rllvm)

m = parseIR("globalString.ll")
ee = ExecutionEngine(m)

.llvm(m[["getStr2"]], .ee = ee)
.llvm(m[["getStr1"]], .ee = ee)

.llvm(m[["setStr1"]], .ee = ee)
.llvm(m[["getStr1"]], .ee = ee)

getGlobalValue(m[["str2_a"]], ee) # works commit: af18e6119
getGlobalValue(m[["str1_p"]], ee) # works now

names(m)
m[["str1"]]
  # works.
getGlobalValue(m[["pi"]], ee)
getGlobalValue(m[["i"]], ee)
getGlobalValue(m[["f"]], ee)

File: explorations/globalString.c
#include <string.h>

char *str1_p = "This is a string";
char str2_a[] = "So is this";
char str3_a_noinit[99];

double pi = 3.141593;
int i = 97;
float f = 34.5;


int
getLen1()
{
    return(strlen(str1_p));
}

int
getLen2()
{
    return(strlen(str2_a));
}

const char * const 
getStr2()
{
    return(str2_a);
}

const char * const 
getStr1()
{
    return(str1_p);
}

void
setStr1()
{
    str1_p = strdup("A replacement string");
}
